payments-ui-core/src/main/java/com/stripe/android/ui/core/PaymentsTheme.kt

```diff
@@ -313,30 +281,75 @@ fun PaymentsTypography.toComposeTypography(): Typography {
     )
 }
 
+val LocalColors = staticCompositionLocalOf { PaymentsTheme.getColors(false) }
+val LocalShapes = staticCompositionLocalOf { PaymentsTheme.shapesMutable }
+val LocalTypography = staticCompositionLocalOf { PaymentsTheme.typographyMutable }
+
+// CAUTION: This theme is mutable by merchant configurations.
 @Composable
 @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
 fun PaymentsTheme(
     content: @Composable () -> Unit
 ) {
-    val colors = PaymentsTheme.colors
-    val localColors = staticCompositionLocalOf { colors }
+    val colors = PaymentsTheme.getColors(isSystemInDarkTheme())
+    val shapes = PaymentsTheme.shapesMutable
+    val typography = PaymentsTheme.typographyMutable
 
-    val shapes = PaymentsTheme.shapes
-    val localShapes = staticCompositionLocalOf { shapes }
+    CompositionLocalProvider(
+        LocalColors provides colors,
+        LocalShapes provides shapes,
+        LocalTypography provides typography
+    ) {
+        MaterialTheme(
+            colors = colors.materialColors,
+            typography = typography.toComposeTypography(),
+            shapes = shapes.toComposeShapes().material,
+            content = content
+        )
+    }
+}
+
+// Use this theme if you do not want merchant configurations to change your UI
```

Comment:
Use Javadoc style for comments in classes/methods, so they're part of docs and the IDE recognizes them (even though they're restricted, still good practice)